legion/engine/core/ecs/ecsregistry.cpp

```diff
@@ -42,8 +42,12 @@ namespace legion::core::ecs
             }
         }
 
-        for (entity_handle& child : data.children)	// Destroy all children.
-            recursiveDestroyEntityInternal(child);
+        if (hasComponent<hierarchy>(entityId))
+        {
+            auto children = entity_handle(entityId).children();
+            for (entity_handle& child : children.reverse_range())	// Destroy all children.
```

Comment:
Not sure how this loops and how children.reverse_range() is retrieved, but it might be faster to save children.reverse_range() and loop through that because the list might have to be removed every iteration